### ISSUE TITLE
Unexpected deprecations when running against Ember 3.27+

### ISSUE THREAD

**Issue by rwjblue** [embroider-build/embroider#738]
The `@embroider/util ember-beta` and `@embroider/util ember-canary` CI jobs are failing due to new deprecations added in Ember 3.27 (currently in beta).

Example failure:

```
not ok 1 Chrome 89.0 - [213 ms] - Integration | Helper | ensure-safe-component: string value
    ---
        actual: >
            Usage of the Ember Global is deprecated. You should import the Ember module or the specific API instead.
        expected: >
            
        stack: >
                at Object.eval (webpack:///./tests/deprecation-assertions.js?:118:12)
                at callHook (http://localhost:7357/assets/test-support.js:5101:28)
                at runHook (http://localhost:7357/assets/test-support.js:5129:11)
                at processTaskQueue (http://localhost:7357/assets/test-support.js:4683:33)
                at advanceTaskQueue (http://localhost:7357/assets/test-support.js:4668:5)
                at Object.advance (http://localhost:7357/assets/test-support.js:4654:5)
        message: >
            Unexpected deprecations
```

**Comment by ef4** [comment]
I ran ember-cli-update in util [here](https://github.com/embroider-build/embroider/pull/740) and fixed the one issue that seems to really be in this package.

There are several others in dependencies that seem like they shouldn't be firing. For example, `ember-resolver/resolvers/classic/index` is throwing the deprecation, even though I can see it's original source uses the Ember import.

Is there some update to the modules-api-polyfill that's missing? Because it did compile that import to a global.

**Comment by ef4** [comment]
Re-rolling the lockfile fixed some of these, but now I think the monorepo is allowing some older version to leak out of the sample apps somehow.

**Comment by rwjblue** [comment]
> There are several others in dependencies that seem like they shouldn't be firing. For example, ember-resolver/resolvers/classic/index is throwing the deprecation, even though I can see it's original source uses the Ember import.

You need ember-resolver@8 (the default blueprint updated to that a while back).

> I ran ember-cli-update in util here and fixed the one issue that seems to really be in this package.

@ef4 - There is still at least one major issue here in Embroider still:

https://github.com/embroider-build/embroider/blob/4ce29ff0da50d4c0bb2d798680427ae2576c4e8e/packages/core/src/babel-plugin-inline-hbs.ts#L117-L118

The babel plugin (https://github.com/ember-cli/babel-plugin-htmlbars-inline-precompile) that is used in classic builds has been updated to account for this (it adds imports instead of the older globals based code), but it seems like we are intentionally duplicating the inline transform here (and I'm not sure why). Do you recall why we needed to duplicate the code? Can we instead add additional options to the main shared babel plugin to do whatever Embroider needs?

**Comment by ef4** [comment]
> Do you recall why we needed to duplicate the code?

The major differences are:

 - we can run in `stage: 1` mode, which only applies custom transforms and re-emits the hbs
 - in `stage: 3` mode, our `TemplateCompiler` does build-time resolution within templates and emits JS imports in addition to just precompiling the HBS.

There's no reason we can't upstream these capabilities into babel-plugin-htmlbars-inilne-precompile, that was just not a priority.

**Comment by ef4** [comment]
This ember-cli-babel bug is causing some of our deprecations: https://github.com/babel/ember-cli-babel/pull/396

**Comment by ef4** [comment]
Also, ember-inspector is causing deprecations.

Also, ember-cli itself is causing one in `contentFor('test-body-footer')`. I see that a newer ember-cli would fix that, but we use the default ember-try config which only updates ember-source, not ember-cli.

**Comment by rwjblue** [comment]
> ember-cli itself is causing one in contentFor('test-body-footer'). I see that a newer ember-cli would fix that, but we use the default ember-try config which only updates ember-source, not ember-cli.

Ya, I’ll be back porting the ember-cli fix to 3.24, 3.25, and 3.26.
